nexus/db-queries/src/db/datastore/saga.rs

```diff
@@ -103,54 +99,4 @@ impl DataStore {
             )),
         }
     }
-
-    pub async fn saga_list_unfinished_by_id(
```

Comment:
Given that both these functions are only called once, and are already part of the `db-queries` crate, I condensed them into `saga_recovery.rs`.

This made it marginally easier to specify arguments, and means we can re-use the same pool connection for each batch we access.

This was mostly for a minor convenience, but I can keep the old structure if you had plans to actually calls these functions from elsewhere.